domain/model/service/service.go

```diff
@@ -542,3 +589,53 @@ func (s *Service) UpdateCredential(
 
 	return s.st.UpdateCredential(ctx, uuid, key)
 }
+
+func getWatchActivatedModelsMapper(st State, logger logger.Logger) func(ctx context.Context, db database.TxnRunner,
+	changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+
+	return func(ctx context.Context, db database.TxnRunner,
+		changes []changestream.ChangeEvent) ([]changestream.ChangeEvent, error) {
+		uuidToChangeEventMap := make(map[string]changestream.ChangeEvent)
+		modelUUIDs := make([]string, len(changes))
+
+		for i, change := range changes {
+			modelUUID := change.Changed()
+			modelUUIDs[i] = modelUUID
+			uuidToChangeEventMap[modelUUID] = change
+		}
+
+		// Retrieve all activate status of all models with associated uuids
+		activatedModelUUIDs, err := st.GetActivatedModelUUIDs(ctx, modelUUIDs)
+
+		// There should be no errors returned if there are no activated models found.
+		if err != nil {
+			logger.Errorf(ctx, "failed to retrieve activated model UUIDs for UUIDs %v: %v", modelUUIDs, err)
+			return nil, err
```

Comment:
You don't need to log and return an error. Do one, not both.

Reply:
I see thanks for the clarification simon!